packages/flutter/lib/src/animation/curves.dart

```diff
@@ -738,9 +738,9 @@ class CatmullRomSpline extends Curve2D {
   Offset transformInternal(double t) {
     _initializeIfNeeded();
     final double length = _cubicSegments.length.toDouble();
-    double position;
-    double localT;
-    int index;
+    final double position;
```

Comment:
`position` is changed in line 745 (4 lines below). How can we make this variable `final`?

Reply:
> `position` is changed in line 745

`position` is rather initialized at line 745. With null-safety you can make a final declaration (the variable will not be initialized at declaration site) and initialize it later. The compiler/analyzer will ensure that when the variable is used it has been fully initialized. So you can have the following initialization:

```dart
final String s;
if (boolCondition) {
  s = 'a';
} else {
  s = 'b';
}
print(s); // if else branch is omitted you will get an error
```

Reply:
ah ha! Thanks for pointing this out. I was not aware of the new "null-safety" support in Dart.